preview/MsixCore/MsixCoreInstallerLib/PackageManager.cpp

```diff
@@ -88,6 +80,70 @@ HRESULT PackageManager::AddPackage(const wstring & packageFilePath, DeploymentOp
     return AddPackage(packageStream.Get(), options);
 }
 
+HRESULT PackageManager::AddPackage(IStream * packageStream, DeploymentOptions options)
+{
+    if (IsWindows10RS3OrLater())
+    {
+        TCHAR tempPathBuffer[MAX_PATH];
+        TCHAR tempFileName[MAX_PATH];
+
+        if (!GetTempPath(MAX_PATH, tempPathBuffer))
+        {
+            return HRESULT_FROM_WIN32(GetLastError());
+        }
+
+        if (!GetTempFileName(tempPathBuffer, TEXT("MSIX"), 0, tempFileName))
+        {
+            return HRESULT_FROM_WIN32(GetLastError());
+        }
+
+        std::wcout << tempFileName << "\n";
+
+        HANDLE tempFileHandle = INVALID_HANDLE_VALUE;
+        tempFileHandle = CreateFile((LPTSTR)tempFileName, GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+        if (tempFileHandle == INVALID_HANDLE_VALUE)
+        {
+            return HRESULT_FROM_WIN32(GetLastError());
+        }
+
+        ULONG dwBytesRead = 0;
+        DWORD dwBytesWritten = 0;
+
+        const int numOfBytes = 2048;
+        BYTE buffer[numOfBytes];
+        HRESULT hr;
+
+        while ((hr = packageStream->Read(buffer, numOfBytes, &dwBytesRead)) == S_OK)
+        {
+            if (dwBytesRead > 0)
+            {
+                if (!WriteFile(tempFileHandle, buffer, dwBytesRead, &dwBytesWritten, NULL))
+                {
+                    return HRESULT_FROM_WIN32(GetLastError());
+                }
+            }
+            else
+            {
+                break;
+            }
+        }
+
+        if (!CloseHandle(tempFileHandle))
+        {
+            return HRESULT_FROM_WIN32(GetLastError());
+        }
+
+        RETURN_IF_FAILED(Windows10Redirector::AddPackage(tempFileName));

```

Comment:
move this into a function inside windows10redirector

Reply:
Refactored to method in windows10redirector.